web/src/components/TreasuryGeneration/NewTreasuryGenerationForm/NewTreasuryGenerationForm.tsx

```diff
@@ -0,0 +1,73 @@
+import { Button } from 'react-bootstrap'
+import { useForm, UseFormReturn } from 'react-hook-form'
+
+import { Form, Label, TextAreaField, Submit } from '@redwoodjs/forms'
+import type { RWGqlError } from '@redwoodjs/forms'
+
+interface CreateNewTreasuryGenerationInput {
+  payload: string
+}
+
+interface NewOrganizationFormProps {
+  onSave: (data: CreateNewTreasuryGenerationInput) => void
+  error: RWGqlError
+  loading: boolean
+}
+
+/* 
+  This form collects any payload information to kick-off a lambda function to generate a treasury file.
+*/
+const NewTreasuryGenerationForm = (props: NewOrganizationFormProps) => {
+  const { onSave, error, loading } = props
+  const formMethods: UseFormReturn = useForm()
+  const hasErrors = Object.keys(formMethods.formState.errors).length > 0
+
+  const onReset = () => {
+    formMethods.reset()
+  }
+
+  const onSubmit = (data) => {
+    const { payload } = data
+
+    onSave({
+      payload,
+    })
+  }
+
+  return (
+    <Form
+      onSubmit={onSubmit}
```

Comment:
Nit: Would it be simpler to just do `() => onSave({ payload })` here rather than creating a new func above?